### PR TITLE
Task1

### PR DESCRIPTION
**Pull request by asutptec4**

### CODE REVIEW

File: src/Point.ts [hunk 1]
```diff
@@ -1 +1,36 @@
...
+  constructor(x?: number, y?: number) {
+    this.x = x ?? 0;
+    this.y = y ?? 0;
+  }
+
+  distance(): number;
+  distance(point: Point): number;
+  distance(x: number, y: number): number;
+  distance(x?: number | Point, y?: number): number {
+    let distance: number;
```
Comment (artabr) [inline src/Point.ts]: Good job!

Just a small comment here. You could've written this structure without an additional variable and else statements:
```
        if (x instanceof Point) {
            return this.calculateDistance(x.x, x.y);
        }
        if (x !== undefined && y !== undefined) {
            return this.calculateDistance(x, y);
        }
        return this.calculateDistance(0, 0);
```
That way the code looks more concise and easy to read IMHO. It's more of the code style preference, and opinionated, so no need to edit this, just FYI.

File: src/Shape.ts [hunk 2]
```diff
@@ -1,3 +1,45 @@
...
+  protected filled: boolean;
+  protected points: Point[];
+
+  constructor(points: Point[]);
+  constructor(points: Point[], color: string, filled: boolean);
+  constructor(points: Point[], color?: string, filled?: boolean) {
+    this.color = color ?? 'green';
+    this.filled = filled ?? true;
+    this.points = points;
+    this.validateShape();
```
Comment (artabr) [inline src/Shape.ts]: If you move the validation before the main code, it will return the error earlier, which is always better.

**Review by artabr — COMMENTED** [review]
You've done great! Please proceed with the next task and feel free to reach me for a Q&A session if you have any questions.

**Comment by asutptec4** [comment]
> You've done great! Please proceed with the next task and feel free to reach me for a Q&A session if you have any questions.

@artabr, thank you for review! I'll begin next task and mark current Q&A as done.

**Pull request closed without merging**